training.py

```diff
@@ -0,0 +1,53 @@
+from argparse import ArgumentParser
+import argparse
+from torch.utils.data import DataLoader
+import torch.optim as optim
+import torch
+from torch.utils.tensorboard import SummaryWriter
+
+
+import data
+from model import Scribe
+import utils
+
+def train(model, dl, num_training_iterations):
+    writer = SummaryWriter()
+    optimizer = optim.Adam(model.parameters())
+    for iteration, batch in enumerate(dl):
+        texts, texts_mask, strokes, strokes_mask = batch
+        if iteration >= num_training_iterations:
+            break
+        optimizer.zero_grad()
+        output = model(strokes)
+
+        # output[0] is the prediction, strokes[1] is the ground truth
+        loss = ((strokes[1:] - output[:-1]) ** 2).mean()
+        if (iteration % 10) == 0:
+            print(f"{loss.item()=:.4f}")
```

Comment:
I encourage you to always track the mean iteration time. As we will be making the model more complex, it's usually a good idea to know how slower we're making the model with the changes that we do.

Reply:
Added mean iteration time.